### DISCUSSION TITLE
🚀 Feature Request: D1 - support Unicode

### DISCUSSION THREAD

**Discussion by GeorgeTailor** [cloudflare/workers-sdk#13495] [Feature Requests]
### Describe the solution

https://www.sqlite.org/faq.html#q18

unicode support is limited, please provide means to use `LOWER`, `UPPER`, `LIKE` on unicode data in D1.

**Comment by Desttro** [comment]
I'm posting here in case it might help - I found a [SQLite ICU (International Components for Unicode) extension](https://www.sqlite.org/src/artifact?ci=trunk&filename=ext/icu/README.txt) that supports case mapping for the entire Unicode character range for SQLite operations/functions.

and [https://github.com/nalgeon/sqlean](https://github.com/nalgeon/sqlean)
